### DISCUSSION TITLE
vs code missing autocompletion

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#27450] [Q&A Tools] [answered]
Dear all,
after upgrading the moose-language extension to the latest version, I lost the autocompletion functionalities (at least for moose style input files). 
Some debugging helped me find this error 

> [Error - 9:41:35 AM] Client MOOSE Language Server: connection to server is erroring.
Header must provide a Content-Length property.

However, I am not sure how to solve it. 

Before, I managed to get it running by downgrading the version of the extension, so now it also throws an error. 

Hope anyone could help. 
Thanks,
mauro

**Comment by mcacace** [comment]
Sorry, but somehow copy and paste did not work properly. This is the full error message

> {"\n\n*** warning, this code is deprecated and will be removed in future versions":"Please update your main.C to adapt new main function in MOOSE framework, see'test/src/main.C in MOOSE as an example of moose::main()'. \nStack frames: 6\n0: libMesh::print_trace(std::ostream&)\n1: void moose::internal::mooseDeprecatedStream<libMesh::BasicOStreamProxy<char, std::char_traits<char> >, char const (&) [136]>(libMesh::BasicOStreamProxy<char, std::char_traits<char> >&, bool, bool, char const (&) [136])\n2: AppFactory::createAppShared(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, char**, int)\n3: main\n4: __libc_start_main\n5: /home/mauro/projects/golem_gitlab/golem-opt(+0x33ef) [0x560f515f13ef]\n\nContent-Length: 286"}
Shutting down server.
[Error - 9:41:35 AM] Stopping server failed
Error: Client is not running and can't be stopped. It's current state is: starting
	at Ed.shutdown (/home/mauro/.vscode/extensions/danielschwen.moose-language-support-1.1.2/out/main.js:41:8468)
	at Ed.stop (/home/mauro/.vscode/extensions/danielschwen.moose-language-support-1.1.2/out/main.js:41:8047)
	at Ed.stop (/home/mauro/.vscode/extensions/danielschwen.moose-language-support-1.1.2/out/main.js:44:38372)
	at Ed.handleConnectionError (/home/mauro/.vscode/extensions/danielschwen.moose-language-support-1.1.2/out/main.js:43:63)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

**Comment by mcacace** [comment]
me again, sorry... I think I solved the issue, by updating the main.C, though I unsure if I am doing it right, and if this was really the problem (parsing?!)

old syntax:
```c++
MooseInit init(argc, argv);
GolemApp::registerApps();
std::shared_ptr<MooseApp> app = AppFactory::createAppShared("GolemApp", argc, argv);
app->run();
``` 
new syntax:
```c++
Moose::main<GolemApp>(argc, argv);
```

**Reply by GiudGiud** [reply]
Hello

this looks right

**Reply by brandonlangley** [reply, marked as answer]
@mcacace -

This issue with deprecated code killing the VS Code extension was also reported yesterday, and I am in the process of fixing it.

The use of deprecated features is currently emitting a warning that breaks the extension integration when it should be ignored.

So if your update does not fully address the issue, then a new MOOSE should be available in the next couple of days fixing this.

**Reply by mcacace** [reply]
@GiudGiud thanks! And, @brandonlangley that's good news. Will wait for the next release, and I'll close the issue.

**Reply by brandonlangley** [reply]
@mcacace -

Just FYI, a fix for this issue was merged into MOOSE yesterday and has already made it to the `devel` and `master` branches.

So deprecated warning messages should no longer kill the VS Code extension if your MOOSE contains this commit: https://github.com/idaholab/moose/commit/6bb1d92f62545bf931dcca4e284cb70ccb959dd7

**Reply by mcacace** [reply]
Excellent (as always). Thanks for reporting @brandonlangley
